### PR TITLE
chore(load-balancer): Add nil pointer checks and tests for the output…

### PR DESCRIPTION
**Pull request by Fyusel**
…Result functions

### CODE REVIEW

File: internal/cmd/load-balancer/list/list_test.go [hunk 1]
```diff
@@ -183,3 +182,30 @@ func TestBuildRequest(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
+		},
```
Comment (rubenhoenle) [inline internal/cmd/load-balancer/list/list_test.go:200]:
```go
{
			name:    "empty loadbalancer in loadbalancers slice",
			args:    args{
                                  loadbalancers: []loadbalancer.LoadBalancer{{}},
                        },
			wantErr: false,
},
```

File: internal/cmd/load-balancer/observability-credentials/add/add_test.go [hunk 2]
```diff
@@ -193,3 +192,38 @@ func TestBuildRequest(t *testing.T) {
...
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: true,
+		},
+		{
+			name: "only credentials response as argument",
+			args: args{
+				resp: &loadbalancer.CreateCredentialsResponse{Credential: &loadbalancer.CredentialsResponse{}},
```
Comment (rubenhoenle) [inline internal/cmd/load-balancer/observability-credentials/add/add_test.go:215]:
```suggestion
				resp: &loadbalancer.CreateCredentialsResponse{},
```

Maybe add a test case which also covers this case

File: internal/cmd/load-balancer/observability-credentials/describe/describe_test.go [hunk 3]
```diff
@@ -211,3 +210,37 @@ func TestBuildRequest(t *testing.T) {
...
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: true,
+		},
+		{
+			name: "only credentials as argument",
+			args: args{
+				credentials: &loadbalancer.GetCredentialsResponse{Credential: &loadbalancer.CredentialsResponse{}},
```
Comment (rubenhoenle) [inline internal/cmd/load-balancer/observability-credentials/describe/describe_test.go:232]:
```suggestion
				credentials: &loadbalancer.GetCredentialsResponse{},
```

same here

File: internal/cmd/load-balancer/observability-credentials/list/list_test.go [hunk 4]
```diff
@@ -269,3 +268,30 @@ func TestGetFilterOp(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
+		},
```
Comment (rubenhoenle) [inline internal/cmd/load-balancer/observability-credentials/list/list_test.go:286]: maybe add a test case which passes an empty credential inside the credentials slice

File: internal/cmd/load-balancer/quota/quota.go [hunk 5]
```diff
@@ -86,6 +86,9 @@ func buildRequest(ctx context.Context, model *inputModel, apiClient *loadbalance
 }
 
 func outputResult(p *print.Printer, outputFormat string, quota *loadbalancer.GetQuotaResponse) error {
+	if quota == nil {
```
Comment (rubenhoenle) [inline internal/cmd/load-balancer/quota/quota.go:89]: seems like the default in this `outputFormat` func is json output and not the pretty output, could be relevant for tests. please check.

**Review by rubenhoenle — APPROVED** [review]

File: internal/cmd/load-balancer/quota/quota.go (hunk 5 above)
Comment (Fyusel) [reply internal/cmd/load-balancer/quota/quota.go:89]: Good catch!
